internal/pkg/core/plugin/process/start.go

```diff
@@ -168,14 +174,14 @@ func injectLogLevel(logLevel string, args []string) []string {
 	var level string
 	// Translate to level beat understands
 	switch logLevel {
-	case "info":
-		level = "info"
-	case "debug":
-		level = "debug"
-	case "warning":
-		level = "warning"
-	case "error":
-		level = "error"
+	case levelInfo:
+		level = levelInfo
+	case levelDebug:
+		level = levelDebug
+	case levelWarning:
+		level = levelWarning
+	case levelError:
+		level = levelError
 	}
 
 	if args == nil || level == "" {
```

Comment:
[Suggestion]
As you're already changing the functions, and considering the addition of a default case, this if could be moved to the very top of the function and deal only with the `args == nil` case. 

Even thought, ignoring the log level when `args == nil` seems to me not quite right form a design perspective. If this function _injects_, instead of just parsing, the log level, it should do so regardless of the value of `args`. If `args` being nil or empty means no log level should be added to it, the caller should check it and then decide if it wants to add the log level or not.

To be honest, I'd change this function to `parseLogLevel`, making it return the _beats compatible_ log level and do the append when calling it:

```
spec.Args = append(spec.Args, parseLogLevel(a.logLevel, spec.Args))
```